### PR TITLE
MySQL SSL & read_timeout options, and bind blob result bugfix

### PR DESCRIPTION
**Pull request by andr-04**
Some useful features for MySQL connector.

### CODE REVIEW

File: include/sqlpp11/mysql/bind_result.h [hunk 1]
```diff
@@ -318,6 +318,9 @@ namespace sqlpp
       void _post_bind_text_result(size_t /* index */, const char** /* text */, size_t* /* len */)
```
[the remark this answers is not in the document]
Comment (andr-04) [reply include/sqlpp11/mysql/bind_result.h:318]: Without this I get the following error:

```
In file included from sqlpp11/include/sqlpp11/data_types/text.h:35,
                 from sqlpp11/include/sqlpp11/result_row.h:31,
                 from sqlpp11/include/sqlpp11/cte.h:34,
                 from sqlpp11/include/sqlpp11/with.h:41,
                 from sqlpp11/include/sqlpp11/sqlite3/serializer.h:43,
                 from sqlpp11/include/sqlpp11/sqlite3/connection.h:566,
                 from sqlpp11/include/sqlpp11/sqlite3/sqlite3.h:30,
                 from db.h:12,
                 from main.cpp:12:
sqlpp11/include/sqlpp11/data_types/text/result_field.h: In instantiation of ‘void sqlpp::result_field_t<Db, sqlpp::field_spec_t<NameType, sqlpp::text, CanBeNull> >::_post_bind(Target&, size_t) [with Target = sqlpp::mysql::bind_result_t; Db = sqlpp::mysql::connection_base; NameType = db::TgSecretChats_::Data::_alias_t; bool CanBeNull = false; size_t = long unsigned int]’:
sqlpp11/include/sqlpp11/result_row.h:71:9:   required from ‘void sqlpp::detail::result_field<Db, index, FieldSpec>::_post_bind(Target&) [with Target = sqlpp::mysql::bind_result_t; Db = sqlpp::mysql::connection_base; long unsigned int index = 2; FieldSpec = sqlpp::field_spec_t<db::TgSecretChats_::Data::_alias_t, sqlpp::text, false>]’
sqlpp11/include/sqlpp11/result_row.h:116:68:   required from ‘void sqlpp::detail::result_row_impl<Db, sqlpp::detail::index_sequence<Ints ...>, FieldSpecs ...>::_post_bind(Target&) [with Target = sqlpp::mysql::bind_result_t; Db = sqlpp::mysql::connection_base; long unsigned int ...Is = {0, 1, 2}; FieldSpecs = {sqlpp::field_spec_t<db::TgSecretChats_::TgSessionId::_alias_t, sqlpp::unsigned_integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::SecretChatId::_alias_t, sqlpp::integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::Data::_alias_t, sqlpp::text, false>}]’
sqlpp11/include/sqlpp11/result_row.h:191:24:   required from ‘void sqlpp::result_row_t<Db, FieldSpecs>::_post_bind(Target&) [with Target = sqlpp::mysql::bind_result_t; Db = sqlpp::mysql::connection_base; FieldSpecs = {sqlpp::field_spec_t<db::TgSecretChats_::TgSessionId::_alias_t, sqlpp::unsigned_integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::SecretChatId::_alias_t, sqlpp::integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::Data::_alias_t, sqlpp::text, false>}]’
sqlpp11/include/sqlpp11/mysql/bind_result.h:92:11:   required from ‘void sqlpp::mysql::bind_result_t::next(ResultRow&) [with ResultRow = sqlpp::result_row_t<sqlpp::mysql::connection_base, sqlpp::field_spec_t<db::TgSecretChats_::TgSessionId::_alias_t, sqlpp::unsigned_integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::SecretChatId::_alias_t, sqlpp::integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::Data::_alias_t, sqlpp::text, false> >]’
sqlpp11/include/sqlpp11/result.h:78:7:   required from ‘sqlpp::result_t<DbResult, ResultRow>::result_t(sqlpp::result_t<DbResult, ResultRow>::db_result_t&&, const DynamicNames&) [with DynamicNames = sqlpp::no_name_t; DbResult = sqlpp::mysql::bind_result_t; ResultRow = sqlpp::result_row_t<sqlpp::mysql::connection_base, sqlpp::field_spec_t<db::TgSecretChats_::TgSessionId::_alias_t, sqlpp::unsigned_integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::SecretChatId::_alias_t, sqlpp::integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::Data::_alias_t, sqlpp::text, false> >; sqlpp::result_t<DbResult, ResultRow>::db_result_t = sqlpp::mysql::bind_result_t]’
sqlpp11/include/sqlpp11/prepared_select.h:50:60:   required from ‘sqlpp::result_t<decltype (db.run_prepared_select((*(const sqlpp::prepared_select_t<Database, Statement, Composite>*)this))), typename Statement::_result_row_t<Database> > sqlpp::prepared_select_t<Database, Statement, Composite>::_run(Database&) const [with Database = sqlpp::mysql::connection_base; Statement = sqlpp::statement_t<void, sqlpp::no_with_t, sqlpp::select_t, sqlpp::no_select_flag_list_t, sqlpp::select_column_list_t<void, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::TgSessionId>, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::SecretChatId>, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::Data> >, sqlpp::from_t<void, db::TgSecretChats>, sqlpp::where_t<void, sqlpp::binary_expression_t<sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::TgSessionId>, sqlpp::op::equal_to, sqlpp::unsigned_integral_operand> >, sqlpp::no_group_by_t, sqlpp::no_having_t, sqlpp::no_order_by_t, sqlpp::no_limit_t, sqlpp::no_offset_t, sqlpp::no_union_t, sqlpp::no_for_update_t>; Composite = sqlpp::statement_t<void, sqlpp::no_with_t, sqlpp::select_t, sqlpp::no_select_flag_list_t, sqlpp::select_column_list_t<void, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::TgSessionId>, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::SecretChatId>, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::Data> >, sqlpp::from_t<void, db::TgSecretChats>, sqlpp::where_t<void, sqlpp::binary_expression_t<sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::TgSessionId>, sqlpp::op::equal_to, sqlpp::unsigned_integral_operand> >, sqlpp::no_group_by_t, sqlpp::no_having_t, sqlpp::no_order_by_t, sqlpp::no_limit_t, sqlpp::no_offset_t, sqlpp::no_union_t, sqlpp::no_for_update_t>; typename Statement::_result_row_t<Database> = sqlpp::result_row_t<sqlpp::mysql::connection_base, sqlpp::field_spec_t<db::TgSecretChats_::TgSessionId::_alias_t, sqlpp::unsigned_integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::SecretChatId::_alias_t, sqlpp::integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::Data::_alias_t, sqlpp::text, false> >; decltype (db.run_prepared_select((*(const sqlpp::prepared_select_t<Database, Statement, Composite>*)this))) = sqlpp::mysql::bind_result_t]’
sqlpp11/include/sqlpp11/mysql/connection.h:424:28:   required from ‘decltype (t._run((*(sqlpp::mysql::connection_base*)this))) sqlpp::mysql::connection_base::_run(const T&, sqlpp::consistent_t) [with T = sqlpp::prepared_select_t<sqlpp::mysql::connection_base, sqlpp::statement_t<void, sqlpp::no_with_t, sqlpp::select_t, sqlpp::no_select_flag_list_t, sqlpp::select_column_list_t<void, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::TgSessionId>, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::SecretChatId>, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::Data> >, sqlpp::from_t<void, db::TgSecretChats>, sqlpp::where_t<void, sqlpp::binary_expression_t<sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::TgSessionId>, sqlpp::op::equal_to, sqlpp::unsigned_integral_operand> >, sqlpp::no_group_by_t, sqlpp::no_having_t, sqlpp::no_order_by_t, sqlpp::no_limit_t, sqlpp::no_offset_t, sqlpp::no_union_t, sqlpp::no_for_update_t>, sqlpp::statement_t<void, sqlpp::no_with_t, sqlpp::select_t, sqlpp::no_select_flag_list_t, sqlpp::select_column_list_t<void, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::TgSessionId>, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::SecretChatId>, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::Data> >, sqlpp::from_t<void, db::TgSecretChats>, sqlpp::where_t<void, sqlpp::binary_expression_t<sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::TgSessionId>, sqlpp::op::equal_to, sqlpp::unsigned_integral_operand> >, sqlpp::no_group_by_t, sqlpp::no_having_t, sqlpp::no_order_by_t, sqlpp::no_limit_t, sqlpp::no_offset_t, sqlpp::no_union_t, sqlpp::no_for_update_t> >; decltype (t._run((*(sqlpp::mysql::connection_base*)this))) = sqlpp::result_t<sqlpp::mysql::bind_result_t, sqlpp::result_row_t<sqlpp::mysql::connection_base, sqlpp::field_spec_t<db::TgSecretChats_::TgSessionId::_alias_t, sqlpp::unsigned_integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::SecretChatId::_alias_t, sqlpp::integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::Data::_alias_t, sqlpp::text, false> > >]’
sqlpp11/include/sqlpp11/mysql/connection.h:433:70:   required from ‘decltype (((sqlpp::mysql::connection_base*)this)->sqlpp::mysql::connection_base::_run(t, sqlpp::run_check_t<sqlpp::mysql::context_t, T>{})) sqlpp::mysql::connection_base::operator()(const T&) [with T = sqlpp::prepared_select_t<sqlpp::mysql::connection_base, sqlpp::statement_t<void, sqlpp::no_with_t, sqlpp::select_t, sqlpp::no_select_flag_list_t, sqlpp::select_column_list_t<void, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::TgSessionId>, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::SecretChatId>, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::Data> >, sqlpp::from_t<void, db::TgSecretChats>, sqlpp::where_t<void, sqlpp::binary_expression_t<sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::TgSessionId>, sqlpp::op::equal_to, sqlpp::unsigned_integral_operand> >, sqlpp::no_group_by_t, sqlpp::no_having_t, sqlpp::no_order_by_t, sqlpp::no_limit_t, sqlpp::no_offset_t, sqlpp::no_union_t, sqlpp::no_for_update_t>, sqlpp::statement_t<void, sqlpp::no_with_t, sqlpp::select_t, sqlpp::no_select_flag_list_t, sqlpp::select_column_list_t<void, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::TgSessionId>, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::SecretChatId>, sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::Data> >, sqlpp::from_t<void, db::TgSecretChats>, sqlpp::where_t<void, sqlpp::binary_expression_t<sqlpp::column_t<db::TgSecretChats, db::TgSecretChats_::TgSessionId>, sqlpp::op::equal_to, sqlpp::unsigned_integral_operand> >, sqlpp::no_group_by_t, sqlpp::no_having_t, sqlpp::no_order_by_t, sqlpp::no_limit_t, sqlpp::no_offset_t, sqlpp::no_union_t, sqlpp::no_for_update_t> >; decltype (((sqlpp::mysql::connection_base*)this)->sqlpp::mysql::connection_base::_run(t, sqlpp::run_check_t<sqlpp::mysql::context_t, T>{})) = sqlpp::result_t<sqlpp::mysql::bind_result_t, sqlpp::result_row_t<sqlpp::mysql::connection_base, sqlpp::field_spec_t<db::TgSecretChats_::TgSessionId::_alias_t, sqlpp::unsigned_integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::SecretChatId::_alias_t, sqlpp::integral, false>, sqlpp::field_spec_t<db::TgSecretChats_::Data::_alias_t, sqlpp::text, false> > >]’
db.h:72:25:   required from here
sqlpp11/include/sqlpp11/data_types/text/result_field.h:63:14: error: ‘class sqlpp::mysql::bind_result_t’ has no member named ‘_post_bind_text_result’; did you mean ‘_post_bind_blob_result’?
   63 |       target._post_bind_text_result(index, &text, &len);
      |       ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
      |       _post_bind_blob_result

```

[the remark this answers is not in the document]
Comment (rbock) [reply include/sqlpp11/mysql/bind_result.h:318]: Ooops, yes. Sorry for the noise...

**Comment by rbock** [comment]
Thanks for the PR. Looks good to me.

github says: "This branch cannot be rebased due to conflicts"? Can you look into that?

**Comment by andr-04** [comment]
> Thanks for the PR. Looks good to me.
> 
> github says: "This branch cannot be rebased due to conflicts"? Can you look into that?

It was a merge conflict which I have resolved yesterday. Can you give more details?

**Comment by rbock** [comment]
See screenshot. That's all I have right now.
image

That said, I can do squash an merge, it seems (that's probably the right approach anyways in this case). I'll leave this open for now in case you want to figure out what caused the message in the screenshot.

**Comment by andr-04** [comment]
Hm... Interesting. Maybe I do merge with a wrong branch (I use main as a base branch)?
image

**Comment by rbock** [comment]
Given that we both have no idea and squashing is an option, I'll just do that :-)

Thanks for the PR!

**Pull request merged**